File: android/gemcore/src/testFixtures/kotlin/com/gemwallet/android/testkit/TransactionExtendedMock.kt
package com.gemwallet.android.testkit

import com.wallet.core.primitives.Asset
import com.wallet.core.primitives.Transaction
import com.wallet.core.primitives.TransactionExtended

fun mockTransactionExtended(
    transaction: Transaction = mockTransaction(),
    asset: Asset = mockAsset(
        id = transaction.assetId,
    ),
    feeAsset: Asset = asset,
    assets: List<Asset> = listOf(asset),
) = TransactionExtended(
    recordId = 1,
    transaction = transaction,
    asset = asset,
    feeAsset = feeAsset,
    price = null,
    feePrice = null,
    assets = assets,
    prices = emptyList(),
    confirmationEtaSeconds = null,
)

File: core/crates/primitives/src/testkit/transaction_extended_mock.rs
use crate::{Asset, Transaction, TransactionExtended};

impl TransactionExtended {
    pub fn mock() -> Self {
        Self::mock_transaction(Transaction::mock())
    }

    pub fn mock_transaction(transaction: Transaction) -> Self {
        let asset = Asset::from_chain(transaction.asset_id.chain);
        TransactionExtended {
            record_id: 1,
            transaction,
            fee_asset: asset.clone(),
            asset,
            price: None,
            fee_price: None,
            assets: vec![],
            prices: vec![],
            from_address: None,
            to_address: None,
            confirmation_eta_seconds: None,
        }
    }
}
